core/src/docs/rfcs/5444_operator_from_uri.md

```diff
@@ -0,0 +1,135 @@
+- Proposal Name: `operator_from_uri`
+- Start Date: 2024-12-23
+- RFC PR: [apache/opendal#5444](https://github.com/apache/opendal/pull/5444)
+- Tracking Issue: [apache/opendal#5445](https://github.com/apache/opendal/issues/5445)
+
+# Summary
+
+This RFC proposes adding URI-based configuration support to OpenDAL, allowing users to create operators directly from URIs. The proposal introduces a new `from_uri` API in both the `Operator` and `Configurator` traits, along with an `OperatorRegistry` to manage operator factories.
+
+# Motivation
+
+Currently, creating an operator in OpenDAL requires explicit configuration through builder patterns. While this approach provides type safety and clear documentation, it can be verbose and inflexible for simple use cases. Many storage systems are naturally identified by URIs (e.g., `s3://bucket/path`, `fs:///path/to/dir`).
+
+Adding URI-based configuration would:
+
+- Simplify operator creation for common use cases
+- Enable configuration via connection strings (common in many applications)
+- Make OpenDAL more approachable for new users
+- Allow dynamic operator creation based on runtime configuration
+
+# Guide-level explanation
+
+The new API allows creating operators directly from URIs:
+
+```rust
+// Create an operator using URI
+let op = Operator::from_uri("s3://my-bucket/path", vec![
+    ("endpoint".to_string(), "http://localhost:8080"to_string()),
```

Comment:
Will the `uri` supports passing config in query string? just like https://www.prisma.io/dataguide/postgresql/short-guides/connection-uris#specifying-additional-parameters

Reply:
Yep, will update the proposal with an example.

Reply:
Added in [`9763a6e` (#5444)](https://github.com/apache/opendal/pull/5444/commits/9763a6e4ac7c0c203a4dd2ef179fa26f801b9bef)

Please take another look, thanks!